easybuild/easyblocks/generic/bundle.py

```diff
@@ -161,36 +162,31 @@ def __init__(self, *args, **kwargs):
             for key in comp_specs:
                 comp_cfg[key] = comp_specs[key]
 
-            # enable resolving of templates for component-specific EasyConfig instance,
-            # but don't require that all template values can be resolved at this point;
-            # this is important to ensure that template values like %(name)s and %(version)s
+            # Don't require that all template values can be resolved at this point but still resolve them.
+            # This is important to ensure that template values like %(name)s and %(version)s
             # are correctly resolved with the component name/version before values are copied over to self.cfg
-            comp_cfg.enable_templating = True
-            comp_cfg.expect_resolved_template_values = False
-
-            # 'sources' is strictly required
-            comp_sources = comp_cfg['sources']
-            if comp_sources:
-                # If per-component source URLs are provided, attach them directly to the relevant sources
-                comp_source_urls = comp_cfg['source_urls']
-                if comp_source_urls:
-                    for source in comp_sources:
-                        if isinstance(source, str):
-                            self.cfg.update('sources', [{'filename': source, 'source_urls': comp_source_urls[:]}])
-                        elif isinstance(source, dict):
-                            # Update source_urls in the 'source' dict to use the one for the components
-                            # (if it doesn't already exist)
-                            if 'source_urls' not in source:
-                                source['source_urls'] = comp_source_urls[:]
-                            self.cfg.update('sources', [source])
-                        else:
-                            raise EasyBuildError("Source %s for component %s is neither a string nor a dict, cannot "
-                                                 "process it.", source, comp_cfg['name'])
-                else:
-                    # add component sources to list of sources
-                    self.cfg.update('sources', comp_sources)
-            else:
+            with comp_cfg.allow_unresolved_templates():
+                comp_sources = comp_cfg['sources']
+            if not comp_sources:
                 raise EasyBuildError("No sources specification for component %s v%s", comp_name, comp_version)
+            # If per-component source URLs are provided, attach them directly to the relevant sources
+            comp_source_urls = comp_cfg['source_urls']
```

Comment:
@Flamefire This should also be done under `allow_unresolved_templates`?

Reply:
Probably yes. Done